browser/src/layer/tile/CommentListSection.ts

```diff
@@ -1123,11 +1123,21 @@ export class CommentSection extends CanvasSectionObject {
 	public adjustParentRemove (comment: any): void {
 		var parentIdx = this.getIndexOf(comment.sectionProperties.data.parent);
 
+		// If a child comment is removed.
 		var parentComment = this.sectionProperties.commentList[parentIdx];
-		if (parentComment && parentComment.sectionProperties.children.includes(comment.sectionProperties.data.id)) {
-			var index = parentComment.sectionProperties.children.indexOf(comment.sectionProperties.data.id);
+		if (parentComment && parentComment.sectionProperties.children.includes(comment)) {
+			var index = parentComment.sectionProperties.children.indexOf(comment);
 			parentComment.sectionProperties.children.splice(index, 1);
 		}
+
+		// If a parent comment is removed.
+		for (var i = 0; i < comment.sectionProperties.children.length; i++) {
+			if (comment.sectionProperties.children[i]) {
+				comment.sectionProperties.children[i].sectionProperties.data.parent = '0';
+				if (this.sectionProperties.docLayer._docType === 'text')
+					comment.sectionProperties.children[i].sectionProperties.data.parentId = '0';
```

Comment:
I would really like to have some method with self explaining human-readable name: `comment.sectionProperties.children[i].setAsRootComment()` or whatewer it does. Annotation code is hard to understand due to magic numbers.

Reply:
Also helper `comment.sectionProperties.children[i]` -> `comment.getChildByIndex(i)` would be nice.

Reply:
Also we can store reference and reuse it. Code is shorter.
```
var child = comment.getChildByIndex(i);
if (child) {
        child.setSomethin();
}
```